### PR TITLE
Core/Guild: Implement new guild bank swap item opcodes

### PR DESCRIPTION
**Pull request by Luzifix**
**Changes proposed:**

- Add new guild bank swap item structs
- Implement guild bank swap item system

**Target branch(es):** master

- [ ] 3.3.5
- [x] master

**Tests performed:**

- [x] It builds
- [x] Tested in-game

### CODE REVIEW

**Review by zTerragor — APPROVED** [review]

**Comment by illfated** [comment]
Happy to see your name still around, @Luzifix , I remember it from my newbie days on GitHub.

File: src/server/game/Server/Protocol/Opcodes.h [hunk 1]
```diff
@@ -365,6 +365,15 @@ enum OpcodeClient : uint16
...
     CMSG_GUILD_BANK_TEXT_QUERY                        = 0x3088,
+    CMSG_GUILD_BANK_TRANSFER_ITEM_TO_BANK             = 0x34BD,
+    CMSG_GUILD_BANK_TRANSFER_ITEM_TO_PLAYER           = 0x34BE,
+    CMSG_GUILD_BANK_MOVE_ITEM_IN_BANK                 = 0x34C1,
+    CMSG_GUILD_BANK_COMBINE_ITEM_STACK_WITH_BANK      = 0x34C2,
+    CMSG_GUILD_BANK_TRANSFER_ITEM_STACK_TO_BANK       = 0x34C3,
+    CMSG_GUILD_BANK_COMBINE_ITEM_STACK_WITH_PLAYER    = 0x34C4,
+    CMSG_GUILD_BANK_TRANSFER_ITEM_STACK_TO_PLAYER     = 0x34C5,
+    CMSG_GUILD_BANK_TRANSFER_ITEM_TO_PLAYER_AUTOSTORE = 0x34C6,
+    CMSG_GUILD_BANK_COMBINE_ITEM_STACK_IN_BANK        = 0x34C7,
```
Comment (ZaDarkSide) [inline src/server/game/Server/Protocol/Opcodes.h:376]: There seems to be missing an opcode here with value 0x34C8

Comment (Luzifix) [reply src/server/game/Server/Protocol/Opcodes.h:376]: I could not trigger the opcode so I just left it out^^

Comment (MaxtorCoder) [reply src/server/game/Server/Protocol/Opcodes.h:376]: `CMSG_GUILD_BANK_MOVE_ITEMS_BANK_BANK_COUNT` is `0x34C8`

Comment (Luzifix) [reply src/server/game/Server/Protocol/Opcodes.h:376]: And what is MOVE_ITEMS_BANK_BANK_COUNT for?

Comment (ZaDarkSide) [reply src/server/game/Server/Protocol/Opcodes.h:376]: image
image
image

I saw this implementation floating around

Comment (MaxtorCoder) [reply src/server/game/Server/Protocol/Opcodes.h:376]: > And what is MOVE_ITEMS_BANK_BANK_COUNT for?

No idea, that's how I have it referenced in my IDB

Comment (Shauren) [reply src/server/game/Server/Protocol/Opcodes.h:376]: There should be 12 of these opcodes total, I can only see 9 in this PR

Comment (ZaDarkSide) [reply src/server/game/Server/Protocol/Opcodes.h:376]: > There should be 12 of these opcodes total, I can only see 9 in this PR

_Playing Gotta catch em' all Pokémon Theme Song_

**Pull request merged**